mig/install/generateconfs.py

```diff
@@ -355,8 +355,8 @@ def usage(options):
         if val == 'DEFAULT':
             del settings[key]
     conf = generate_confs(**settings)
-    # print "DEBUG: %s" % conf
-    instructions_path = "%(destination)s/instructions.txt" % conf
+    # TODO: avoid reconstructing this path (also done inside generate_confs)
+    instructions_path = "%s/instructions.txt" % conf['destination_path']
```

Comment:
Fine with the removal of 'DEBUG' print and the addition of the TODO. Any particular motivation for switching from the `'%(destination)s' % conf` form to the `'%s % conf['destination_path']` one?

Reply:
Yah - what this was doing was trying to write the file through the symlink to the confs directory, but the effect of doing so is identical to writing the instructions to the suffixed output directory given which is still accessible via the symlink. Doing so was important for some of the codepaths that test the symlink, but also quietly sets up being able to symlink _after_ we successfully write the whole config in the future.

Reply:
Oh, and up at this level, the value _within_ the function can have been changed (e.g. a suffix added after applying `keyword_auto` to the destination) so we take the calculated path we _know_ was just written to and use it here rather then duplicate a bunch of logic. I will say this is another one of those looks slightly odd due to trying to minimize change - my hope is land the tests then revisit :)

Reply:
Right, I think I can follow your logic and it makes sense. I mainly wondered if you had a motivated preference for `'%s' % conf['val']` over the interchangeable but IMO slightly less clunky `'%(val)s' % conf` form, but I guess not then :-)

Reply:
Ohhh right - I didn't quite know the latter was a thing! I guess my initial initial reaction is it feels a little magic, but can also see that if you were doing multiple replacements based on a dict there would be some nice properties to that. In this one-off case I'd lean to leave it clunky haha but yeah, no real preference :-)